contracts/DomainStorage.sol

```diff
@@ -0,0 +1,114 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.0;
+
+/// @title Domain controller
+/// @author Serhii Bondarenko
+/// @notice You can use this contract for storing a domain name and an address of the domain owners
+/// @dev The contract owner can change the price of domains and withdraw ether from the contract
+/// @custom:experimental This is an experimental contract.
+contract DomainStorage {
+
+    address public owner;
+    uint256 public domainCost;
+    Domain[] public domains;
+
+    /// @notice Event during the registration of a new domain
+    event DomainAdded(
+        address indexed from,
+        uint indexed when,
+        string domain,
+        address domainOwner
+    );
+
+    /// @notice Domain structure that contains a domain name and the owner address
+    struct Domain {
+        string domainName;
+        address domainOwner;
+    }
+
+    /// @notice Create a new domain storage contract
+    /// @param _domainCost The price of each domain in Ether
+    /// @dev Set deployer as the contract owner
+    constructor(uint256 _domainCost) {
+        owner = msg.sender;
+        domainCost = _domainCost;
+    }
+
+    /// @notice Modifier to check for contract owner
+    modifier onlyOwner() {
+        require(msg.sender == owner, "Not contract owner");
+        _;
+    }
+
+    /// @notice Modifier to check that provided domain can be only first level
+    modifier checkSingleDomain(string memory domain) {
+        bytes memory dotBytes = bytes (".");
+        bytes memory domainBytes = bytes (domain);
+
+        require(domainBytes.length >= dotBytes.length);
+
+        bool found = false;
```

Comment:
// line 50 
// bool fount by default is false, it is not necessary to set it to false
// - 1 point